stdlib/2and3/shutil.pyi

```diff
@@ -41,36 +58,41 @@ else:
     def copyfile(src: _Path, dst: _Path) -> None: ...
     def copymode(src: _Path, dst: _Path) -> None: ...
     def copystat(src: _Path, dst: _Path) -> None: ...
-    def copy(src: _Path, dst: _Path) -> None: ...
-    def copy2(src: _Path, dst: _Path) -> None: ...
+    def copy(src: _Path, dst: _Path) -> _PathReturn: ...
+    def copy2(src: _Path, dst: _Path) -> _PathReturn: ...
 
-def ignore_patterns(*patterns: _Path) -> Callable[[_Path, List[str]],
-                                                  Iterable[str]]: ...
+def ignore_patterns(*patterns: _Path) -> Callable[[Any, List[_AnyStr]], Set[_AnyStr]]: ...
 
-_IgnoreFn = Union[None, Callable[[str, List[str]], Iterable[str]], Callable[[_Path, List[str]], Iterable[str]]]
-if sys.version_info >= (3, 3):
+if sys.version_info >= (3,):
+    _IgnoreFn = Union[None, Callable[[str, List[str]], Iterable[str]], Callable[[_Path, List[str]], Iterable[str]]]
     def copytree(src: _Path, dst: _Path, symlinks: bool = ...,
                  ignore: _IgnoreFn = ...,
                  copy_function: Callable[[str, str], None] = ...,
                  ignore_dangling_symlinks: bool = ...) -> _PathReturn: ...
 else:
-    def copytree(src: str, dst: str, symlinks: bool = ...,
-                 ignore: _IgnoreFn = ...,
-                 copy_function: Callable[[str, str], None] = ...,
-                 ignore_dangling_symlinks: bool = ...) -> None: ...
-
-def rmtree(path: _Path, ignore_errors: bool = ...,
-           onerror: Callable[[Any, Any, Any], None] = ...) -> None: ...
+    _IgnoreFn = Union[None, Callable[[AnyStr, List[AnyStr]], Iterable[AnyStr]]]
+    def copytree(src: AnyStr, dst: AnyStr, symlinks: bool = ...,
+                 ignore: _IgnoreFn = ...) -> _PathReturn: ...
+
+if sys.version_info >= (3,):
+    @overload
+    def rmtree(path: bytes, ignore_errors: bool = ...,
+               onerror: Callable[[Any, str, Any], None] = ...) -> None: ...
```

Comment:
The return type of the callback should be `Any` for flexibility (for example, a user could pass in a lambda that accidentally returns some value).